packages/trip-details/src/index.tsx

```diff
@@ -0,0 +1,272 @@
+import flatten from "flat";
+import { MessageDescriptor } from "@formatjs/intl";
+// @ts-expect-error FIXME: Create TypeScript types for core-utils packages.
+import coreUtils from "@opentripplanner/core-utils";
+import moment from "moment";
+import React, { ReactElement } from "react";
+import { defineMessages, FormattedMessage, FormattedNumber } from "react-intl";
+import { CalendarAlt, Heartbeat, MoneyBillAlt } from "styled-icons/fa-solid";
+
+import * as Styled from "./styled";
+import TripDetail from "./trip-detail";
+import { CaloriesDetailsProps, TripDetailsProps } from "./types";
+
+// Load the default messages.
+import defaultEnglishMessages from "../i18n/en-US.yml";
+
+// HACK: We should flatten the messages loaded above because
+// the YAML loaders behave differently between webpack and our version of jest:
+// - the yaml loader for webpack returns a nested object,
+// - the yaml loader for jest returns messages with flattened ids.
+const defaultMessages: Record<string, string> = flatten(defaultEnglishMessages);
+
+/**
+ * Construct full formatjs message definitions and pass them to defineMessages.
+ */
+// TODO: Find a better place for this utility.
+function expandAndDefineMessages(
+  prefix: string,
+  entries: Record<string, string>
+): Record<string, MessageDescriptor> {
+  const messages: Record<string, MessageDescriptor> = {};
+  Object.keys(entries).forEach((key: string) => {
+    const id = `${prefix}.${key}`;
+    messages[key] = {
+      defaultMessage: defaultMessages[id],
+      description: entries[key],
+      id
+    };
+  });
+  return defineMessages(messages);
+}
+
+// Internationalization messages definition. This is used for tooling purposes.
+// Message descriptions and default messages are included so they can be captured
+// with react-intl CLI and documentation.
+// See https://formatjs.io/docs/react-intl/api#definemessagesdefinemessage
+const messages = expandAndDefineMessages("otpUi.TripDetails", {
+  calories:
+    "Text showing the number of calories for the walking and biking legs of a trip.",
+  caloriesDescription:
+    "Text describing how the calories relate to the walking and biking duration of a trip.",
+  departure: "Text showing the departure date/time for a trip.",
+  title: "Title (heading) text of the component.",
+  tncFare: "Text showing the price paid to transportation network companies.",
+  transitFare: "Text showing the price paid for public transit tickets."
+});
```

Comment:
I went ahead and tried using the `@formatjs/cli` cli tool and have verified that this approach is not able to properly generate a list of messages because that tool does not appear to do any runtime analysis of what is imported from the `../i18n/en-US.yml` file. It only really seemed to work if message strings were explicitly spelled out as full strings without any prefix or dynamic help. 

Therefore, I'm not convinced this is the exemplar approach that we should be using in every otp-ui file moving forward. The reason I used the `react-intl#defineMessages` method in otp-rr was because I thought if I didn't obtain that variable from the props in a functional component like in [this example](https://formatjs.io/docs/tooling/cli) that it wouldn't work. However, the cli tool does appear to pick up a signature of `intl.formatMessage({ id: "someId" })` anywhere, so I think the `defineMessages` thing is only useful when the same message ID would be used in multiple `intl.formatMessage` calls.

So, after this analysis, my recommendation is as follows:

- go ahead and still import the default messages, and I guess we have to flatten them because of jest (although it'd be ideal if we didn't have to)
- Remove the `expandAndDefineMessages` function and import of `react-intl#defineMessages`.
- In each FormattedMessage component 
  - spell out the entire flattened message ID in the `id` prop of a `FormattedMessage` 
  - spell out the entire flattened message ID in a lookup of the default message from the imported flattened default messages
  - if we are to keep the descriptions, also add a description prop

Reply:
Thanks for clarifying the recommendations, I made the changes in https://github.com/opentripplanner/otp-ui/pull/281/commits/5f530377e60008a64f3408968fab91ec4f02e380.

Reply:
`@formatjs/cli` is interesting to add, I'll try that in a separate PR.

Reply:
https://github.com/ibi-group/otp-ui/pull/9 has a basic message id comparison between code and ymls using `@formatjs/cli`.